salt/renderers/gpg.py

```diff
@@ -37,6 +37,56 @@
     increasing the system entropy. On virtualised Linux systems, this can often
     be achieved by installing the ``rng-tools`` package.
 
+Import keys to a master
+************************
+
+If the keys already exist and need to be imported to the salt master, run the
+following to import them.
+
+.. code-block:: bash
+
+    gpg --import /path/to/private.key
+    gpg --import /path/to/pubkey.gpg
+
+If the salt master runs as normal user, become this user before importing the
+keys. The default target dir will be ``~/.gnupg``. This can be overridden by
+the ``--homedir`` option. The keys must be at least readable for the runuser.
+
+Adjust trust level of imported keys
+***********************************
+
+In some cases, importing existing keys may not be enough an the trust level of
+the key needs to be adjusted. This can be done by editing the key. The ``key_id``
+and the actual trust level of the key can be seen by listing the already imported
+keys.
+
+.. code-block:: bash
+
+    gpg --list-keys
+    gpg --list-secret-keys
+
+If the trust-level is not ``ultimate`` it needs to be changed by running
+
+.. code-block:: bash
+
+    gpg --edit-key <key_id>
+
+This will open an interactive shell for the management of the GPG encrypted key. Type
+``trust`` to be able to set the trust level for the key and then select
+``5 (I trust ultimately)``. Then quit the shell by typing ``save``.
+
+Enable usage of GPG keys on the master
+**************************************
+
```

Comment:
This section is misleading. The ``gpg_keydir`` setting is only required if a non-standard path is used.

Reply:
Wontfix:

The salt tutorial requires one to put some keys into a non-standard location and then assumes this is the default location for gpg keys. If the salt-master is running as non-root user it would be fair to assume that the default dir is `/path/to/salt-master-homedir/.gpg`.

So, the default location is misleading if not outright wrong and the default value is documented as use case only. Now it becomes at least clear that the option exists and can be used. I assume people will be smart enough to omit the option, if it is not needed (documented location used).

Reply:
The existing documentation uses ``/etc/salt/gpgkeys``. This is also the default location. I'm not sure how you can say this is either misleading or "outright wrong" when it's a well-documented and expected location for these keys.

This section is written as if the option is required, particularly the part where the heading is, "Enable usage of GPG keys on the master". This section should likely be titled something closer to, "Using a different path for keys" and written from that perspective.

Reply:
The existing documentation uses `/etc/salt/gpgkeys`, which is a non-standard location when it comes to gpg keys (one has to explicitly use the `homedir` option).
So, when it comes to standard locations, the default path should be `~/.gpg`. 

I have no idea why the gpg key handling with salt is so over complicated. People may already have (local) gpg keys and are used to the default gpg key location. 

Plus: Having the keys in /etc/ requires me to track another dir for correct permissions if I don't want to run the salt-master as root user.